### ISSUE TITLE
Make sure our existing generators are up-to-date

### ISSUE THREAD

**Issue by petertseng** [exercism/xgo#604]
In [x-common](https://github.com/exercism/x-common/tree/master/exercises), we have various exercises with canonical-data.json representing the recommended set of tests for that exercise. We created generators to more easily keep up with changes in x-common. Whenever x-common changes, the task is as simple as re-running the generator.

The current documentation is at https://github.com/exercism/xgo#generating-test-cases - if at any point we find that we should explain something better, we should add any necessary documentation.

Preliminaries:

* I find the list of our generators using `ls exercises/*/cases_test.go | cut -d/ -f2` from the root of the xgo directory.
* I find the list of exercises with canonical-data.json using `ls exercises/*/canonical-data.json | cut -d/ -f2` from the root of the x-common directory.

This issue concerns **exercises that already have a generator**.

Because of recent changes to make all the canonical-data.json files conform to a common scheme (https://github.com/exercism/x-common/issues/625), it might be the case that our current generator no longer runs.

For these exercises we'd make sure the current generator runs, or if it doesn't then we'd update it.

In addition, the three lines at the top of the file:

```
// Source: {{.Ori}}
{{if .Commit}}// Commit: {{.Commit}}
{{end}}
```

can simply be replaced with `{{.Header}}`, which has the effect of:

* Deprecating the `Ori` field, which is confusingly named
* Showing the x-common version in the test file
* Reducing duplicate code

## Relevant exercises

- bob
- bowling
- change
- clock (#620)
- connect
- custom-set (in progress #622)
- forth (#615)
- gigasecond (#619)
- hamming
- largest-series-product
- leap (#606)
- luhn (#618)
- meetup
- raindrops
- rna-transcription
- roman-numerals
- transpose
- word-count

Once all these are complete, we can delete the deprecated `Ori` field.

**Comment by leenipper** [comment]
> recent changes to make all the canonical-data.json files 

The forth exercise generator became askew due to that.  It runs, but test cases are empty. I have a WIP for it.

**Comment by robphoenix** [comment]
I've changed this issue from a checklist to multiple issues. I've left the links to relevant PR's in the original comment but I don't think it's necessary to continue adding these.

Let me know if any of this is wrong/not as good as it could be! BTW [Hub](https://hub.github.com/) was really useful for this. I was able to repeat the command `git issue create -l "good first patch","#604 confirm generators are up-to-date"` and it would open my editor with the same message so I just needed to change the exercise in the title. Still a little repetitive, maybe if my bash skills were better... :smile:
